x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearningUsageTransportAction.java

```diff
@@ -441,39 +443,28 @@ private void addTrainedModelStats(GetTrainedModelsAction.Response response, Map<
     }
 
     // TODO separate out ours and users models possibly regression vs classification
-    private void addInferenceIngestUsage(NodesStatsResponse response, Map<String, Object> inferenceUsage) {
-        Set<String> pipelines = new HashSet<>();
+    private void addInferenceIngestUsage(GetTrainedModelsStatsAction.Response statsResponse, Map<String, Object> inferenceUsage) {
+        int pipelineCount = 0;
         Map<String, Long> docCountStats = new HashMap<>(3);
         Map<String, Long> timeStats = new HashMap<>(3);
         Map<String, Long> failureStats = new HashMap<>(3);
         initializeStats(docCountStats);
         initializeStats(timeStats);
         initializeStats(failureStats);
 
-        response.getNodes()
```

Comment:
The original reason behind this was to prevent the 10k limit we currently have on stats and configs. It is conceivable that a user has 10k+ models, none of which are deployed and then have some later that are deployed + referenced in ingest pipelines.

Since we are already limiting our DFA collection to the first 10k, this is probably acceptable.